File: 2020_year/12_month/3_week/jihun/algo/1260.js
let fs = require('fs');
let input = fs.readFileSync('./dev/stdin').toString().split('\n');

let [N, M, V] = input[0].split(' ').map(el => +el);
let flag = []
let edge = []
let check = []

for (let i = 0; i < N; i++) {
    flag.push(0)
}

for (let i = 0; i < N+1; i++) {
    let temp = []
    let temp2 = []
    for (let j = 0; j < N+1; j++) {
        temp.push(0)
        temp2.push(0)
    }
    edge.push(temp)
    check.push(temp2)
}

for (let i = 1; i < M+1; i++) {
    let [node1, node2] = input[i].split(' ').map(el => +el)
    edge[node1][node2] = 1
    edge[node2][node1] = 1
}

let dfsAnswer = []
const dfs = function(n, v) {
    if (flag.every(node => node === 1)) {
        return
    }
    if (flag[v-1] === 0) {
        flag[v-1] = 1
        dfsAnswer.push(v)
    }
    for (let i = 1; i < n+1; i++) {
        if (edge[v][i] === 1 && check[v][i] !== 1) {
            check[v][i] = 1
            dfs(n, i)
        }
    }
}
dfs(N, V)
console.log(dfsAnswer)

File: 2020_year/12_month/3_week/jihun/algo/1260.py
import sys
sys.stdin = open("./dev/stdin")

N, M, V = list(map(int, input().split()))

flag = [1] + [0] * (N)
edge = [[0] * (N + 1) for _ in range(N+1)]

for i in range(M):
    s, g = list(map(int, input().split()))
    edge[s][g] = 1
    edge[g][s] = 1

dfs_answer = []
def dfs(n, v):
    if (0 not in flag):
        return
    if (flag[v] == 0):
        flag[v] = 1
        dfs_answer.append(v)
    else:
        return
    for i in range(1, n+1):
        if edge[v][i] == 1 and flag[i] == 0:
            dfs(n, i)
dfs(N, V)

flag = [1] + [0] * (N)
check = [[0] * (N + 1) for _ in range(N+1)]

bfs_answer = []
queue = []
def bfs(n, v):
    queue = [v]
    while queue:
        stack = queue.pop(0)
        flag[stack] = 1
        bfs_answer.append(stack)
        if (0 not in flag):
            return
        for i in range(1, n+1):
            if (edge[stack][i] and not flag[i]):
                queue.append(i)

bfs(N, V)

print(*dfs_answer)
print(*bfs_answer)